src/libs/elektra/keyhelpers.c

```diff
@@ -190,12 +190,17 @@ int keyNameIsSystem (const char * name)
 /**
  * @internal
  *
+ * @pre key->meta must be NULL or a valid keyset
```

Comment:
Why do we need this special code for key->meta? Why not always clearing it and creating it new?

Reply:
The main problem is that `keyInit` is called twice during the creation of a key *with a name*. Once in `elektraKeyMalloc` and then again in `keyVInit`. I tried removing the second occurence but it failed for some tests so I assumed it was necessary and didn't look into it further. I totally agree that the meta keyset initialization is rather ugly.

Due to this, `key->meta` might be a valid keyset when `keyInit` is called a second time. If I always clear the key, without a check, then that's a memory leak.

Since `keyInit` is the actual place where the keys struct is initialized to 0, I have to set `key->meta = NULL` even before that so that I know whether that is the first or second call.
Hope that makes sense.

Reply:
I think you can inline and remove `elektraKeyMalloc`. It is only used twice and only once the call to `keyInit` is required (when called in `keyNew`).